pkg/context/detached_test.go

```diff
@@ -0,0 +1,61 @@
+//go:build test_unit
+
+package context
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/nuclio/logger"
+	nucliozap "github.com/nuclio/zap"
+	"github.com/stretchr/testify/suite"
+)
+
+type DetachedTestSuite struct {
+	suite.Suite
+	logger logger.Logger
+}
+
+func (suite *DetachedTestSuite) SetupTest() {
+	suite.logger, _ = nucliozap.NewNuclioZapTest("test")
+}
+
+func (suite *DetachedTestSuite) TestDetachedOnCancelledParent() {
+	timerChannel := make(chan int)
+	expectedWaitTime := 5
+
+	parentCtx, cancelParentCtx := context.WithCancel(context.Background())
+	childCtx, cancelChildCtx := context.WithCancel(NewDetached(parentCtx))
+
+	go suite.testChildNotCancelled(childCtx, timerChannel)
+
+	suite.logger.Debug("Cancelling parent context")
+	cancelParentCtx()
+	time.Sleep(6 * time.Second)
+	suite.logger.Debug("Cancelling child context")
+	cancelChildCtx()
+
+	suite.Require().Equal(expectedWaitTime, <-timerChannel)
+}
+
+func (suite *DetachedTestSuite) testChildNotCancelled(ctx context.Context, ch chan int) {
```

Comment:
I wouldn't call this method with this name, as you should reserve `test` prefixes for methods which are actually tests. And also this method doesn't do what its name says it does